### PR TITLE
Refator the LibevdevWrapper

### PR DESCRIPTION
**Pull request by ndesh26**
I have created a temporary commit which moves the common methods of `Device` and `UninitDevice` into the `LibevdevWrapper`. I want to finalize the name for the trait and the structs. `LibevdevWrapper` seems a bit wordy to me. I also want to finalize the location of the code (maybe have a separate file for the trait itself). All of these decisions will affect the API exposed to the user. Since this release will be breaking the programs with the most common scenarios. I wanted to get it right. 

Please share any suggestions regarding the update.  @maxbla

### CODE REVIEW

**Comment by maxbla** [comment]
Let me describe a problem, then my limited understanding of how to fix it. When we switch from dedicated functions to traits, users must `use evdev-rs::Trait` to call any of the evdev functions (note: I think this omission is why this PR is failing). It's not clear to the user why the traits need to exist as first glance, and traits make the methods harder to find in the API docs. We're moving further in this direction - #42 mentions making enable generic, using a trait.

One way to fix this is with a prelude, like the [one for std](https://doc.rust-lang.org/std/prelude/index.html). The prelude would probably just contain both traits - no need to `use Device` and potentially cause a naming conflict. In all out example code, we would add an `import evdev-rs::prelude::*` to demonstrate how to properly use it.

Another way to deal with it is to just have both pub methods and a pub Trait that exposes the same methods. I don't like this, as it is repetitive.

Re: naming, `LibevdevWrapper` could be renamed to `LibevdevDevice`, `DeviceWrapper` (or `DevWrapper`). I feel like it should be called `Device`, but that's already a struct name, so it would be too confusing.

> I also want to finalize the location of the code (maybe have a separate file for the trait itself)

> All of these decisions will affect the API exposed to the user.

The location of the code and how a user `ues`s it are disconnected because re-exports can hide the true location from the user. The trait should be in the top-level namespace of this crate ([example](https://doc.rust-lang.org/reference/items/use-declarations.html#use-visibility)) regardless of where the code lives. On this subject, I don't think the enums module should be exposed to the user - we should probably be re-exporting everything from enums.rs in lib.rs, but that doesn't *need* to be done for this release.

File: src/uinput.rs [hunk 1]
```diff
@@ -15,6 +15,10 @@ pub struct UInputDevice {
 unsafe impl Send for UInputDevice {}
 
 impl UInputDevice {
+    fn raw(&self) -> *mut raw::libevdev_uinput {
```
Comment (maxbla) [inline src/uinput.rs:18]: I don't understand why we need this method. Isn't it redundant with `<UinputDevice as LibevdevWrapper>::raw()`?

Comment (maxbla) [reply src/uinput.rs:18]: The same question applies to most methods in this file

Comment (maxbla) [reply src/uinput.rs:18]: Ignore this comment - I thought LibevdevWrapper was supposed to abstract over `UInputDevice`s in addition to `UninitDevice`s and `Device`s

**Comment by maxbla** [comment]
I think EvdevWrapper (or whatever it is renamed to) should be exposed at the crate root, not device, so consumers of this API `use evdev::EvdevWrapper`. Slap a pub(crate) on the declaration in device.rs and a pub use in lib.rs. Also I now favor `DeviceWrapper` and `DeviceLike` as names for it.

**Comment by ndesh26** [comment]
I don't understand your comment fully. I have re-exported the `EvdevWrapper` in `lib.rs` already. Can you elaborate more on your comment? 

I picked the name `EvdevWrapper` because the trait itself doesn't carry any data about a device. It acts as a collection of methods to interact with an evdev device. These are those methods that could be executed irrespective of the presence of a file being attached to the device.

**Comment by maxbla** [comment]
My mistake. I think my confusing comment applied to the state of the code when I wrote #60.

As for the naming, either work, but I think EvdevWrapper is worse than LibevdevWrapper, since evdev is a kernel interface and libevdev is a userspace program. I mantain that my suggestion for DeviceLike makes sense, and is more beginner-friemdly. `DeviceLike` carries a pointer to a struct that represents an opened device, so it does carry information about a device (just behind an opaque struct)

**Comment by ndesh26** [comment]
Let's use `DeviceWrapper` then. It is less ambiguous than `EvdevWrapper`.
